scapy/layers/stamp.py

```diff
@@ -0,0 +1,334 @@
+       1: the party generating the timestamp has a clock that
+          is synchronized to UTC using an external source
+
+    Z is interpreted as follows (defined in Section 2.3 of [RFC 8186]):
+        0: NTP 64-bit format of a timestamp
+        1: PTPv2 truncated format of a timestamp
+
+    Scale and Multiplier are linked by the following relationship:
+        ErrorEstimate = Multiplier*2^(-32)*2^Scale (in seconds)
+
+
+    References:
+        * https://www.rfc-editor.org/rfc/rfc4656.html
+        * https://www.rfc-editor.org/rfc/rfc8186.html
+    """
+
+    name = 'Error Estimate'
+    fields_desc = [
+        BitEnumField('S', 0, 1, _sync_types),
+        BitEnumField('Z', 0, 1, _timestamp_types),
+        BitField('scale', 0, 6),
+        ByteField('multiplier', 1),
+    ]
+
+    def guess_payload_class(self, payload):
+        # type: (str) -> Packet_metaclass
+        # Trick to tell scapy that the remaining bytes of the currently
+        # dissected string is not a payload of this packet but of some other
+        # underlayer packet
+        return config.conf.padding_layer
+
+
+class STAMPTestTLV(Packet):
+    """
+    The STAMP Test TLV defined in Section 4 of [RFC 8972] provides a flexible
+    extension mechanism for optional informational elements.
+
+    The TLV Format in a STAMP Test packet is reported below:
+
+        0                   1                   2                   3
+        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |STAMP TLV Flags|     Type      |           Length              |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       ~                            Value                              ~
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    STAMP TLV Flags: 8-bit field; for the details about the STAMP TLV
+                     Flags Format, see RFC 8972
+
+    Type: characterizes the interpretation of the Value field
+
+    Length: the length of the Value field in octets
+
+    Value: interpreted according to the value of the Type field
+
+
+    References:
+        * https://www.rfc-editor.org/rfc/rfc8972.html
+    """
+
+    name = 'STAMP Test Packet - Generic TLV'
+    fields_desc = [
+        FlagsField('flags', 0, 8, "UMIRRRRR"),
+        ByteEnumField('type', None, _stamp_tlvs),
+        ShortField('len', 0),
+        StrLenField('value', '', length_from=lambda pkt: pkt.len),
+    ]
+
+    def extract_padding(self, p):
+        return b"", p
+
+    registered_stamp_tlv = {}
+
+    @classmethod
+    def register_variant(cls):
+        cls.registered_stamp_tlv[cls.type.default] = cls
+
+    @classmethod
+    def dispatch_hook(cls, pkt=None, *args, **kargs):
+        if pkt:
+            tmp_type = ord(pkt[:1])
+            return cls.registered_stamp_tlv.get(tmp_type, cls)
+        return cls
+
+
+class Timestamp(Packet):
```

Comment:
We have a field for that ! You can use https://github.com/secdev/scapy/blob/799f272bc04c361841d01e9c0087950e0eb86610/scapy/fields.py#L3328 with `fmt="Q"`, it'll be more user-friendly than a PacketField over this

Reply:
Thanks for the suggestion. I updated the PR to use the UTCTimeField.